src/test/shell/bazel/remote/remote_execution_test.sh

```diff
@@ -3480,44 +3502,45 @@ platform(
     },
 )
 
-sh_test(
+my_test(
     name = "test",
-    srcs = ["test.sh"],
+    # TODO: This uses a hack by setting test.no-remote-exec on the exec platform
+    # forced by this constraint for both the build and the test action. Instead,
+    # use exec_group_compatible_with = {"test": [":has_foo"]} once it is
+    # implemented.
     exec_compatible_with = [":has_foo"],
 )
 
-sh_test(
+my_test(
     name = "test2",
-    srcs = ["test.sh"],
-    exec_compatible_with = [":has_foo"],
-    target_compatible_with = [":has_foo"],
+    exec_properties = {
+        "test.no-remote-exec": "1",
+    },
```

Comment:
Nit: I think it would be better to keep the old test setup and add new targets for `my_test` rule. That would show that the old behavior is unchanged for `sh_test`(?)

Reply:
`sh_test` no longer lives in the Bazel codebase and isn't versioned together with Bazel. Since this test necessarily depends on internals of the rule (the actions it registers), it wouldn't be self-contained if it used that rule. 

In fact, the test in its previous state relied on `sh_test` to register a "build action" that is also backed by a spawn, but `sh_test` uses `ctx.actions.symlink`, which is spawn-less. This hid the bug that the different test spawns used different execution infos. Using a custom rule defined within the test avoids this class of issues.